common/lib/xmodule/xmodule/js/src/video/00_sjson.js

```diff
@@ -27,11 +27,27 @@ function() {
             return sjson.text.length;
         };
 
-        var search = function (time) {
-            var start = getStartTimes(),
+        var search = function (time, startTime, endTime) {
+           var start,
                 max = size() - 1,
                 min = 0,
-                index;
+                index,
+                results;
+
+            /* if we specify a start and end time to search,
+             * search the filtered list of captions in between
+             * the start / end times.
+             * Else, search the unfiltered list.
+             */
+            if (typeof startTime != 'undefined' &&
+                typeof endTime != 'undefined' &&
```

Comment:
The first undefined check is for arity, since this function can take either 1 or 3 parameters.
The second check is for the 3 parameter case; we check whether endTime is a legal value. 
This assumes some defaults:

- startTime is an integer >0
- endTime is an integer or `null` if it's an illegal value. 

Since these defaults are enforced outside the function, I'm tempted to comment rather than enforce these rules in js, since this function is used only once in its 3-param form in `06_video_captions.js`, but I can see a reason for doing so explicitly.